### ISSUE TITLE
Nunjucks parameters with the same name produce duplicate captions/IDs

### ISSUE THREAD

**Issue by querkmachine** [alphagov/govuk-design-system#2260]
## Description of the issue
Currently, if a Nunjucks parameter has nested child parameters, the parent parameter is automatically split into its own table in the Nunjucks docs, captioned 'Options for [parameter]', and given an ID in HTML similar to `[example ID]--[parameter]`. 

If two or more Nunjucks parameters share the same name—which is possible if they are themselves nested under separate parent parameters—then multiple tables with the same caption and ID are generated.

## Steps to reproduce the issue
1. Go to https://design-system.service.gov.uk/components/footer/
2. Expand the 'Nunjucks' tab of the first example.
3. Expand the 'Nunjucks macro options' section of the Nunjucks tab.
4. Inspect the tables nested within the macro options.

## Actual vs expected behaviour

Actual:
* There are two tables captioned 'Options for items'.
* Both tables have the ID `options-default-1--items`.
* The duplicate IDs means that the 'See items' link within the `navigation` options links to the wrong table. 

Expected: 
* It should be clear that one of these tables refers to the `items` child parameter of `meta`, and the other to the `items` child parameter of `navigation`.
* As table captions may be used to identify each table, they should ideally have unique captions. 
* IDs should be unique to conform with [WCAG criterion 4.1.1](https://www.w3.org/WAI/WCAG21/Understanding/parsing).

**Comment by colinrotherham** [comment]
@querkmachine @dav-idc I've picked this one up, fixed in:

* https://github.com/alphagov/govuk-design-system/pull/3107
